pkg/chunk/cache/instrumented.go

```diff
@@ -39,23 +21,33 @@ var (
 		// 1024 * 4^(7-1) = 4MB
 		Buckets: prometheus.ExponentialBuckets(1024, 4, 7),
 	}, []string{"name", "method"})
-)
 
-func init() {
-	requestDuration.Register()
-	prometheus.MustRegister(fetchedKeys)
-	prometheus.MustRegister(hits)
-	prometheus.MustRegister(valueSize)
-}
+	requestDuration := promauto.With(reg).NewHistogramVec(prometheus.HistogramOpts{
```

Comment:
This is problematic because doesn't have `name` constant label, while it should have in order to avoid clashing metrics (which panic at registration). I'm OK adding the `name` here as well, but please mention it as `[CHANGE]` in the changelog.